internal/provider/zfs.go

```diff
@@ -296,14 +538,19 @@ type CreatePool struct {
 func createPool(config *Config, pool *CreatePool) (*Pool, error) {
 	serialized_options := ""
 	for property, value := range pool.properties {
-		serialized_options = fmt.Sprintf(" -o %s=%s", property, value)
+		if isPoolProperty(property) {
+			serialized_options += fmt.Sprintf(" -o '%s'='%s'", property, value)
+		} else {
+			serialized_options += fmt.Sprintf(" -O '%s'='%s'", property, value)
+		}
 	}
 
+	callSshCommand(config, "zpool destroy %s", pool.name) // TODO remove
 	_, err := callSshCommand(config, "zpool create %s %s %s", serialized_options, pool.name, pool.vdevs)
 
 	if err != nil {
 		// We might have an error, but it's possible that the pool was still created
```

Comment:
I've left this intact as I wasn't sure what the motivation for this was, but as far as I can tell this isn't actually a useful action. The logic seems inverted (the fetched pool is returned _only_ if `describePool` returns an error) and the returned pool is never used. Maybe I'm missing something, or maybe the thing that this was used for has changed, but as things stand I think the whole "fetch-after-fail" could be removed entirely (also applies to `createDataset`).

Reply:
I understand the confusion, I believe I experienced I case wherein the pool would get created, but errors still occurred, such as with the underlying SSH connection for example, which means that the pool would exist, but errors still occurred.

The awkward error checking logic is to determine that case.